src/poetry/repositories/legacy_repository.py

```diff
@@ -33,6 +33,10 @@ def __init__(
 
         super().__init__(name, url.rstrip("/"), config, disable_cache)
 
+    @property
+    def packages(self) -> list[Package]:
+        return []
```

Comment:
Why that?

Reply:
Can we add a comment in the code why we are just returning an empty list please?

Reply:
Or a docstring :)

Reply:
Looking into this again, I wonder why we return an empty list indeed. The rationale for changing this is to remove the special casing in `Pool.search`. However, `LegacyRepository.search` only browses `LegacyRepository.packages` and inspects their names (one may debate whether that is 'searching', but it is what happens). As such, it appears that *even if `LegacyRepository.packages` is populated*, this should not be a heavy task. Should we undo the change to `LegacyRepository`? If not, what is the rationale to be provided in the docstring?

Related (though a bit of a sidestep): it appears that the `search` command (console/commands/search) considers `pypi` only and ignores all other repositories (even if `pypi` is disabled). Is that the intended behaviour? It seems odd.

Reply:
It's indeed a good question. I believe currently this is because `poetry search` only ever historically searched PyPI, and when we refactored repositories the first time (the old code was quite jumbled), `search` was implemented on the base `Repository` class even if it doesn't make a ton of sense on `LegacyRepository` as we only populate it on demand.

If we teach the search command to use `LegacyRepository` and use some sort of lightweight `PackageVersion` structure that doesn't require inspection like creating a `Package`, this is probably worth revisiting (see https://github.com/python-poetry/poetry/issues/2446) -- but that's work for another day. For now a regular comment instead of docstring explaining we're stubbing this out to avoid changing behavior/it's a TODO to rethink search so all source types can support it likely makes more sense.